tasks/main.yml: Fix issues running while in `--check` mode

RyanSquared opened the pull request:
Some issues that arose:

- `check` mode disables the ability to use commands w/o `always_run` on.
- `with_items` is evaluated before `when`, meaning if the `ls` command
  isn't run, Ansible will give an error about `keys` not being valid.
  * A better solution to this would be to instead use the `find` module;
    however, as of right now, this will work.

RyanSquared committed:
tasks/main.yml: Fix issues running while in `--check` mode

Some issues that arose:

- `check` mode disables the ability to use commands w/o `always_run` on.
- `with_items` is evaluated before `when`, meaning if the `ls` command
  isn't run, Ansible will give an error about `keys` not being valid.
  * A better solution to this would be to instead use the `find` module;
    however, as of right now, this will work.

RyanSquared committed:
tasks/main.yml: Change always_run for check_mode

daurnimator reviewed (approved):
Sounds okay. I read docs about check_mode here http://docs.ansible.com/ansible/latest/playbooks_checkmode.html

KellerFuchs commented:
IIRC, we didn't use `find` because it is Ansible 2.x only.
This isn't a problem anymore (we require Ansible 2.0 or greater), so I'll send a PR for that.

KellerFuchs commented:
Also, the `authorized_keys` “trick” that `credentials.yml` uses fails with `--check`.
We should make another PR for it (and ideally send a patch upstream to make `authorized_keys` loop-aware).

KellerFuchs committed:
tasks/main: Gate relevant tasks with `when: "{{ exclusive }}"`

As @RyanSquare pointed out, `with_items` is evaluated before `when`,
resulting in an undefined variable in the second task.

KellerFuchs reviewed (approved):
LGTM with the change I pushed in.
I would like @daurnimator and/or someone else to re-review, though, since I changed stuff.

RyanSquared commented:
I'm gonna pull this and check it out, but it looks alright. However, I'm not actually able to submit a review over my own code.

Edit: Works fine. :+1: for merging.
